@types/automerge/index.d.ts

```diff
@@ -0,0 +1,275 @@
+declare module 'automerge' {
+  function applyChanges<T>(doc: T, changes: Change<T>[]): T
+  function canRedo<T>(doc: T): boolean
+  function canUndo<T>(doc: T): boolean
+  function change<T>(doc: T, message: string, callback: ChangeFn<T>): T
+  function change<T>(doc: T, callback: ChangeFn<T>): T
+  function diff<T>(oldDoc: T, newDoc: T): Diff
+  function emptyChange<T>(doc: T, message?: string): T
+  function equals<T>(val1: T, val2: T): boolean
+  function getActorId<T>(doc: T): UUID
+  function getChanges<T>(oldDoc: T, newDoc: T): Change<T>[]
+  function getConflicts<T>(doc: T, key: Key): any
+  function getHistory<T>(doc: T): State<T>[]
+  function getMissingDeps<T>(doc: T): Clock
+  function getObjectById<T>(doc: T, objectId: UUID): T
+  function getObjectId<T>(doc: T): string
+  function init<T>(actorId?: string): T
+  function load<T>(doc: T, actorId?: string): T
+  function merge<T>(localDoc: T, remoteDoc: T): T
+  function redo<T>(doc: T, message?: string): T
+  function save<T>(doc: T): T
+  function setActorId<T>(doc: T, actorId: UUID): T
+  function undo<T>(doc: T, message?: string): T
+  function getElemId<T=string>(object: List<T> | Text, index: number): UUID
+
+  class Connection<T> {
+    constructor(docSet: DocSet<T>, sendMsg: (msg: Message<T>) => void)
+    close(): void
+    docChanged(docId: UUID, doc: T): void
+    maybeSendChanges(docId: UUID): void
+    open(): void
+    receiveMsg(msg: Message<T>): T
+    sendMsg(docId: UUID, clock: Clock, changes: Change<T>[]): void
+  }
+
+  class Table<T, KeyOrder extends Array<keyof T>> {
+    constructor(columns: KeyArray<T, KeyOrder>)
+    [Symbol.iterator](): {
+      next: () => {
+        done: boolean
+        value: T
+      }
+    }
+    add(item: T | TupleFromInterface<T, KeyOrder>): UUID;
+    byId(id: UUID): T
+    columns: string[]
+    count: number
+    ids: UUID[]
+    filter(fn: filterFn<T>): T[]
+    find(fn: filterFn<T>): T
+    map<U>(fn: (elem: T) => U): U[]
+    remove(id: UUID): void
+    rows(): T[]
+    set(id: UUID, value: T): void
+    set(id: 'columns', value: string[]): void
+    sort(arg?: Function | string | string[]): void
+  }
+
+  class List<T> extends Array<T> {
+    insertAt?(index: number, ...args: T[]): List<T>
+    deleteAt?(index: number, numDelete?: number): List<T>
+  }
+
+  class Text extends List<string> {
+    constructor(objectId?: UUID, elems?: string[], maxElem?: number)
+    get?(index: number): string
+    getElemId(index: number): UUID
+  }
+
+  class DocSet<T> {
+    constructor()
+    applyChanges(docId: UUID, changes: Change<T>[]): T
+    getDoc(docId: UUID): T
+    setDoc(docId: UUID, doc: T): void
+    registerHandler(handler: Handler<T>): void
+    unregisterHandler(handler: Handler<T>): void
+  }
+
+  class WatchableDoc<T> {
+    constructor(doc: T)
+    applyChanges(changes: Change<T>[]): T
+    get(): T
+    set(doc: T): void
+    registerHandler(handler: Handler<T>): void
```

Comment:
The `handler` argument here should be a function with signature `(doc: T) => void` (there is no `UUID` argument).

I suggest renaming `Handler` to `DocSetHandler` and introducing a new type `WatchableDocHandler`.

/cc @HerbCaudill